keps/sig-apps/3998-job-success-completion-policy/README.md

```diff
@@ -0,0 +1,847 @@
+Those policies can be modeled in the following:
+
+1. An indexed job completes if a set of [x, y, z...] indexes are successful.
+2. An indexed job completes if x of indexes are successful.
+
+Then, when the job meets one of the success policies, a new condition, `SuccessCriteriaMet,` is added.
+
+### User Stories (Optional)
+
+#### Story 1
+
+As a machine-learning researcher, I run an indexed job which a leader is launched
+as index=0 and workers are launched as index=1+.
+I want to care about only the leaders when the result of job is evaluated.
+
+In addition, we want to terminate the lingering pods if the leader index (index=0) is succeeded
+because the workers often don't have any ways to terminate themselves due to launching daemon processes like ssh-server.
+
+```yaml
+apiVersion: batch/v1
+kind: Job
+spec:
+  parallelism: 10
+  completions: 10
+  completionMode: Indexed
+  successPolicy:
+    criteria:
+    - succeededIndexes: "0"
+  template:
+    spec:
+      restartPolicy: Never
+      containers:
+      - name: job-container
+        image: job-image
+        command: ["./sample"]
+```
+
+#### Story 2
+
+As a simulation researcher/engineer for fluid dynamics/biochemistry.
+I want to mark the job as successful and terminate the lingering pods
+when the job meets the one of following conditions:
+
+1. The case of the leader index (index=0) is succeeded
+2. The case of some worker indexes (index=1+) are succeeded
+
+Because succeeded leader index means that the whole simulation is succeeded,
+and succeeded some worker indexes means that the minimum required value is satisfied.
+
+```yaml
+apiVersion: batch/v1
+kind: Job
+spec:
+  parallelism: 10
+  completions: 10
+  completionMode: Indexed
+  successPolicy:
+    criteria:
+    - succeededCount: 1
+      succeededIndexes: 0
+    - succeededCount: 5
+      succeededIndexes: "1-9"
+  template:
+    spec:
+      restartPolicy: Never
+      containers:
+        - name: job-container
+          image: job-image
+          command: ["./sample"]
+```
+
+#### Story 3
+
+As a developer of CustomJob API built with Job API like JobSet, I want to implement the reconcile logic so that
+the controller can change the actions based on the reason why the job has been succeeded.
+
+So, it should be included in the `reason` field instead of `message` field since the reason field should be machine-readable. 
+
+```yaml
+apiVersion: batch/v1
+kind: Job
+spec:
+  parallelism: 10
+  completions: 10
+  completionMode: Indexed
+  successPolicy:
+    criteria:
+    - succeededIndexes: "0"
+      setSuccessCriterionMetReason: "LeaderSucceeded"
+    - succeededIndexes: "1-9"
+      setSuccessCriterionMetReason: "WorkersSucceeded"
+  template:
+    spec:
+      restartPolicy: Never
+      containers:
+        - name: job-container
+          image: job-image
+          command: ["./sample"]
+status:
+  conditions:
+  - type: "SuccessCriteriaMet"
+    status: "True"
+    reason: "JobSuccessPolicyAsLeaderSucceeded"
```

Comment:
if it's no longer a suffix, this needs updating.

Reply:
Sorry, this is my bad. I will update this.

Reply:
Done.